apps/service_gather/lib/gather/extraction/source_stream.ex

```diff
@@ -8,41 +8,103 @@ defmodule Gather.Extraction.SourceStream do
     """
     use Source.Handler
     use Properties, otp_app: :service_gather
+    use Annotated.Retry
+    require Logger
+
+    @max_tries get_config_value(:max_tries, default: 10)
+    @initial_delay get_config_value(:initial_delay, default: 500)
 
     getter(:dlq, default: Dlq)
+    getter(:app_name, required: true)
 
-    def handle_batch(batch, context) do
-      send(context.assigns.pid, {:source_batch, batch})
-      :ok
+    @impl Source.Handler
+    def handle_batch(batch, %{assigns: %{extract: extract}} = context) do
+      Decoder.decode(extract.decoder, [batch])
+      |> Ok.each(fn chunk ->
+        messages =
+          Enum.map(chunk, &lowercase_fields/1)
+          |> normalize(extract)
+
+        :ok = write_to_destination(extract.destination, context.assigns.destination_pid, messages)
+      end)
+    catch
+      _, e ->
+        warn_extract_failure(extract, e)
+        send(context.assigns.pid, {:extract_failed, e})
```

Comment:
Do we want to stop processing for all problems?  This seems like a situation where the dlq would be employed?

Reply:
I think i understand that u'll only realy ever land in catch when the destination call fails since everythign else just works without exceptions, but I'm curious if that is true

Reply:
I think we need to understand what are the scenarios in which an error is raised.

Reply:
Effectively we were doing this already; if the extract were churning through a stream and failed to write it would kick the entire thing back to the `Extraction.extract` function which would retry a certain number of times and then kill the whole stream. At least now this is a more granular operation on the handling of each individual batch. We should only reach that scenario if for some reason our destination is completely unreachable. (also checked with @jdenen and he agreed the "shut it all down" approach was okay/best)